### PR TITLE
Adapt stats endpoint to read JSONs bigger than 64K

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/17671|

## Description
Adds pagination for `/agents/{agent_id}/stats/{component}` endpoint.

### CODE REVIEW

**Review by mhamra — CHANGES_REQUESTED** [review]
In the function `get_daemons_stats_from_socket` while's cycle

https://github.com/wazuh/wazuh/blob/f30ed6851baaa95da2043dc79fe039d6b00c913c/framework/wazuh/core/stats.py#L359-L380

The if statement has similar statements in the True and False branches. Please refactor to avoid repeating similar code. 

Suggestion:
- you can assign socket_response.get('json_updated', False) to a next_page bool variable, unify all the repeated code and use that in an if to separate the differences.
- You can also change the last `for` cycle by a list comprehension, similar to what is used in line 367.

**Comment by RamosFe** [comment]
Refactored in commit 92d458e7e5bf3525a83e19c1ed427d90867440ea.

**Review by mhamra — CHANGES_REQUESTED** [review]
Please udate the Docstrings in is_agent_a_manager, create_stats_command and check_if_daemon_exists_in_agent.

**Comment by RamosFe** [comment]
Updated docstrings in commit 88370a2161b08b0b3a31192cbc70d41fc469cc2a.

**Review by mhamra — APPROVED** [review]
LGTM!

File: framework/wazuh/core/stats.py [hunk 1]
```diff
@@ -219,59 +219,172 @@ def get_daemons_stats_(filename: str) -> list:
...
+        socket_response = json.loads(rec_msg)
+
+        if last_json_updated:
+            data = socket_response['data']
+            data.update((k, utils.get_utc_strptime(data[k], "%Y-%m-%d %H:%M:%S").strftime(common.DATE_FORMAT))
+                        for k, v in data.items() if k in {'last_keepalive', 'last_ack'})
+        else:
+            page_data = socket_response['data']
+            # Extends existing items
+            [data[key].extend(item) for key, item in page_data.items() if key in data and isinstance(item, list)]
```
Comment (Selutario) [inline framework/wazuh/core/stats.py:388]: This is wrong, the lists are nested in `data.global.files` but the `if` condition is checking whether `data.global` is a list (it is not a list but a dict). The same happens for `data.interval.files`. 

As a consequence, data is not extended and only partial results are being returned.

File: framework/wazuh/core/stats.py [hunk 2]
```diff
@@ -219,59 +219,172 @@ def get_daemons_stats_(filename: str) -> list:
...
+    socket_response = json.loads(socket_msg)
+
+    # Handle error
+    if socket_response.get('error', 0) != 0:
+        rec_msg = socket_response.get('message', "")
         raise WazuhError(1117, extra_message=rec_msg)
+
+    data = socket_response['data']
+    data.update((k, utils.get_utc_strptime(data[k], "%Y-%m-%d %H:%M:%S").strftime(common.DATE_FORMAT))
+                for k, v in data.items() if k in {'last_keepalive', 'last_ack'})
```
Comment (Selutario) [inline framework/wazuh/core/stats.py:372]: Something similar is happening here. Data received looks like:
```JSON
{
  "data": {
    "affected_items": [
      {
        "global": {
          "start": "2023-11-23 10:22:46",
          "end": "2023-11-23 10:51:46",
          "files": [
            {
              "location": "last -n 20",
              "events": 5,
              "bytes": 330,
              "targets": [
                {
                  "name": "agent",
                  "drops": 0
                }
              ]
            },
     ...
```

But the `if` condition inside the loop checks if the `global` key is `last_keepalive` or `last_ack`. Neither `last_keepalive` nor `last_ack` exist in the response. 

Also, at the `global` level there are no date-shaped keys, they are nested in one more key.

**Review by Selutario — CHANGES_REQUESTED** [review]

**Review by mhamra — CHANGES_REQUESTED** [review]
I've tested the agent and manager using the `feature/17671-read-64k-json-stat` branch but couldn't get the test to work.
See my [comment](https://github.com/wazuh/wazuh/pull/18613#issuecomment-1839494773), I think the PR's branch maybe requires a rebase to be in synch with the changes of the core.

**Comment by RamosFe** [comment]
## Update

After a rebase with the base branch, the API worked as expected:
- Master response (Configuration in master node id `000`): [responseMaster.json](https://github.com/wazuh/wazuh/files/13554327/responseMaster.json)
- Agent response (Configuration in agent node id `001`): [responseAgent.json](https://github.com/wazuh/wazuh/files/13554326/responseAgent.json)

**Review by Selutario — APPROVED** [review]

**Comment by mhamra** [comment]
### TESTING UPDATE 
I've repeated the API call [GET /agents/:agent_id/stats/:component](url) test with the following configuration:

```
MANAGER_BRANCH=feature/17671-read-64k-json-stat
MANAGER_VERSION=4.7
AGENT_BRANCH=feature/17671-read-64k-json-stat
AGENT_VERSION=4.7
WAZUH_PYTHON_VERSION=3.9
```

After following the same steps found in this [comment](https://github.com/wazuh/wazuh/pull/18613#issuecomment-1839494773), the API call generated the following response: 

[api-response.json.zip](https://github.com/wazuh/wazuh/files/13566900/api-response.json.zip)

**Pull request merged**